css/how_to_write_responsive_css.md

```diff
@@ -0,0 +1,153 @@
+
+The main tool we use for this task is the CSS media-query function
+
+```scss
+// SCSS example media-query
+.fjord-button {
+    // @media (min-width: 992px) {...}
+    @include media-breakpoint-up(lg) {...} // Bootstrap mixin, recommended
+}
+```
+
+Thanks to the media-queries we are able to set rules that apply only to some devices or screen sizes.
+
+Since Clay is usually the boilerplate for our products, we should use the [provided mixins and breakpoints](https://github.com/liferay/clay/blob/master/packages/clay-css/src/scss/bootstrap/mixins/_breakpoints.scss) to create consistency throughout the whole application.
+
+## 2. What is mobile first?
+
+The term "mobile first" was born in a Visual/UX context where it's extremely important to create the design starting from the mobile version and going up covering all the devices
+
+In CSS, it's ok to start developing from the desktop design, sometimes it has a higher priority or we don't have access to the mobile version
+
+The real problem appears when we start adding CSS rules in order to create the small version of our application.
+
+```scss
+// SCSS example bad
+.fjord-text {
+	font-size: 1.5rem;
+
+	//@media (max-width: 768px) {
+	@include media-breakpoint-down(sm) {
+		font-size: 1rem;
+	}
+}
+```
+
+```scss
+// SCSS example good
+.fjord-text {
+	font-size: 1rem;
+
+	//@media (min-width: 768px) {
+	@include media-breakpoint-up(sm) {
+		font-size: 1.5rem;
+	}
+}
+```
+
+In a standard scenario, desktops have higher rendering capabilities and better internet connections, so they can easily afford more design complexity.
+
+Since small devices need more attentions from developers, we prefer to make the mobile version of our application as simple and light as possible.
+
+To create a good responsive, we should prevent CSS overwrites in the mobile version, a good trick is to avoid using `@media (max-width: value) {` but it's not always valid
+
+!["example 1"](./images/example_1.jpg)
+
+There are 2 ways to obtain that result in the example image
+
+> We are going to omit the code we need to create the button and focus only on the position
+
+```scss
+// SCSS example bad
+.fjord-header-button-plus {
+	position: fixed;
+	right: 2rem;
+	bottom: 2rem;
+	z-index: 10; // z-index should be a variable
+
+	//@media (min-width: 768px) {
+	@include media-breakpoint-up(sm) {
+		position: initial;
+		right: initial;
+		bottom: initial;
+		z-index: initial;
+	}
+}
+```
+
+```scss
+// SCSS example good
+.fjord-header-button-plus {
+	//@media (max-width: 768px) {
+	@include media-breakpoint-down(xs) {
+		position: fixed;
+		right: 2rem;
+		bottom: 2rem;
+		z-index: 10; // z-index should be a variable
+	}
+}
+```
+
+In this example you can see how sometimes we recommend to use `@media (max-width: value) {` when it creates a lighter code
+
+## 3. What about the HTML?
+
+From the beginning CSS and HTML have always been the perfect _couple_, this time is not different.
```

Comment:
Hey @wincnt, does this make sense in english? is there a different saying for it? I know you can say _a match made in heaven_ but not sure if the perfect couple makes sense.

I also think it might be "Since the beginning of time..."?

Reply:
Wearing my English teacher hat, I'd say:

- We shouldn't emphasize "couple" with italics as it carries no special weight in the sentence.
- "this time is no different" would be more idiomatic than "this time is not different".
- You're right that "match made in heaven" is an idiom, but it is probably overstating things a bit 😂 — we could tone this whole thing down a bit without making claims about "From the earliest moments after the Big Bang..."

Reply:
@eduardoallegrini